### PR TITLE
Add security logging

### PR DESCRIPTION
**Pull request by agaricusbisporus**
only BE. logging relevant data for security dep. after entities manipulated

### CODE REVIEW

File: src/main/java/app/coronawarn/quicktest/model/SecurityAuditListenerQuickTest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+@ConfigurationProperties("auditlogs")
+public class SecurityAuditListenerQuickTest {
+
+    private final QuickTestConfig quickTestConfig;
+    private final Utilities utilities;
+    private String pattern = "User: {}; tenantId: {}; pocID: {}; pocName {}; action: {}; Object: {}; ID: {}";
```
Comment (mbaude) [inline src/main/java/app/coronawarn/quicktest/model/SecurityAuditListenerQuickTest.java:23]:
```suggestion
    private String pattern = "User: {}; tenantId: {}; pocID: {}; action: {}; Object: {}; ID: {}";
```

File: src/main/java/app/coronawarn/quicktest/model/SecurityAuditListenerQuickTest.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    @PostLoad
+    private void afterSelectQuickTest(QuickTest quickTest) throws QuickTestServiceException {
+        log.info(pattern,
+                utilities.getUserNameFromToken(),
+                utilities.getIdsFromToken().get(quickTestConfig.getTenantIdKey()),
+                utilities.getIdsFromToken().get(quickTestConfig.getTenantPointOfCareIdKey()),
+                utilities.getIdsFromToken().get(quickTestConfig.getPointOfCareIdName()),
```
Comment (mbaude) [inline src/main/java/app/coronawarn/quicktest/model/SecurityAuditListenerQuickTest.java:31]:
```suggestion
```

File: src/main/java/app/coronawarn/quicktest/model/SecurityAuditListenerQuickTest.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    @PostPersist
+    @PostUpdate
+    private void afterSaveQuickTest(QuickTest quickTest) throws QuickTestServiceException {
+        log.info(pattern,
+                utilities.getUserNameFromToken(),
+                utilities.getIdsFromToken().get(quickTestConfig.getTenantIdKey()),
+                utilities.getIdsFromToken().get(quickTestConfig.getTenantPointOfCareIdKey()),
+                utilities.getIdsFromToken().get(quickTestConfig.getPointOfCareIdName()),
```
Comment (mbaude) [inline src/main/java/app/coronawarn/quicktest/model/SecurityAuditListenerQuickTest.java:44]:
```suggestion
```

File: src/main/java/app/coronawarn/quicktest/model/SecurityAuditListenerQuickTestArchive.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+@ConfigurationProperties("auditlogs")
+public class SecurityAuditListenerQuickTestArchive {
+
+    private final Utilities utilities;
+    private final QuickTestConfig quickTestConfig;
+    private String pattern = "User: {}; tenantId: {}; pocID: {}; pocName {}; action: {}; Object: {}; ID: {}";
```
Comment (mbaude) [inline src/main/java/app/coronawarn/quicktest/model/SecurityAuditListenerQuickTestArchive.java:23]:
```suggestion
    private String pattern = "User: {}; tenantId: {}; pocID: {}; action: {}; Object: {}; ID: {}";
```

File: src/main/java/app/coronawarn/quicktest/model/SecurityAuditListenerQuickTestArchive.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+    @PostLoad
+    private void afterSelectQuickTestArchive(QuickTestArchive quickTestArchive) throws QuickTestServiceException {
+        log.info(pattern,
+                utilities.getUserNameFromToken(),
+                utilities.getIdsFromToken().get(quickTestConfig.getTenantIdKey()),
+                utilities.getIdsFromToken().get(quickTestConfig.getTenantPointOfCareIdKey()),
+                utilities.getIdsFromToken().get(quickTestConfig.getPointOfCareIdName()),
```
Comment (mbaude) [inline src/main/java/app/coronawarn/quicktest/model/SecurityAuditListenerQuickTestArchive.java:31]:
```suggestion
```

File: src/main/java/app/coronawarn/quicktest/model/SecurityAuditListenerQuickTestArchive.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    @PostPersist
+    @PostUpdate
+    private void afterSaveQuickTestArchive(QuickTestArchive quickTestArchive) throws QuickTestServiceException {
+        log.info(pattern,
+                utilities.getUserNameFromToken(),
+                utilities.getIdsFromToken().get(quickTestConfig.getTenantIdKey()),
+                utilities.getIdsFromToken().get(quickTestConfig.getTenantPointOfCareIdKey()),
+                utilities.getIdsFromToken().get(quickTestConfig.getPointOfCareIdName()),
```
Comment (mbaude) [inline src/main/java/app/coronawarn/quicktest/model/SecurityAuditListenerQuickTestArchive.java:44]:
```suggestion
```

File: src/main/java/app/coronawarn/quicktest/model/SecurityAuditListenerQuickTestStatistics.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+@ConfigurationProperties("auditlogs")
+public class SecurityAuditListenerQuickTestStatistics {
+
+    private final QuickTestConfig quickTestConfig;
+    private final Utilities utilities;
+    private String pattern = "User: {}; tenantId: {}; pocID: {}; pocName {}; action: {}; Object: {}; ID: {}";
```
Comment (mbaude) [inline src/main/java/app/coronawarn/quicktest/model/SecurityAuditListenerQuickTestStatistics.java:21]:
```suggestion
    private String pattern = "User: {}; tenantId: {}; pocID: {}; action: {}; Object: {}; ID: {}";
```

File: src/main/java/app/coronawarn/quicktest/model/SecurityAuditListenerQuickTestStatistics.java [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+    @PostLoad
+    private void afterSelectQuickTest(QuickTestStatistics quickTestStatistics) throws QuickTestServiceException {
+        log.info(pattern,
+                utilities.getUserNameFromToken(),
+                utilities.getIdsFromToken().get(quickTestConfig.getTenantIdKey()),
+                utilities.getIdsFromToken().get(quickTestConfig.getTenantPointOfCareIdKey()),
+                utilities.getIdsFromToken().get(quickTestConfig.getPointOfCareIdName()),
```
Comment (mbaude) [inline src/main/java/app/coronawarn/quicktest/model/SecurityAuditListenerQuickTestStatistics.java:29]:
```suggestion
```

File: src/main/resources/application.yml [hunk 9]
```diff
@@ -73,6 +72,9 @@ management:
     mail:
       enabled: off
 
+auditlogs:
+  pattern: "User: {}; tenantId: {}; pocID: {}; pocName {}; action: {}; Object: {}; ID: {}"
```
Comment (mbaude) [inline src/main/resources/application.yml:76]:
```suggestion
  pattern: "User: {}; tenantId: {}; pocID: {}; action: {}; Object: {}; ID: {}"
```

**Review by mbaude — CHANGES_REQUESTED** [review]
remove pocname, that doesn't exist

**Review by bjarne1337 — APPROVED** [review]

**Review by mbaude — APPROVED** [review]

**Pull request merged**